integrations/jta/jdbc/src/main/java/io/helidon/integrations/jta/jdbc/JtaConnection.java

```diff
@@ -1205,6 +1196,7 @@ private Enlistment(Transaction transaction, XAResource xaResource) {
         }
 
         private Enlistment {
+            threadId = Thread.currentThread().getId();
             Objects.requireNonNull(transaction, "transaction");
             Objects.requireNonNull(xaResource, "xaResource");
         }
```

Comment:
Seems like its used on one place only, why not set the threadId there?

Reply:
The thought was:
* `Enlistment` is a record
* `threadId` needs to be part of its equality and hash code contract
* I _could_ override both instead in which case now the "recordness" is kind of pointless
* Meh, it's a `private` internal class anyway so let me just ensure that the thread id is always set correctly

Reply:
In my latest change I went ahead and just simplified all this so you have to specify the thread ID.